lighthouse-core/computed/trace-of-tab.js

```diff
@@ -6,7 +6,34 @@
 'use strict';
 
 const makeComputedArtifact = require('./computed-artifact.js');
-const TraceOfTab_ = require('../lib/tracehouse/trace-of-tab.js');
+const LHError = require('../lib/lh-error.js');
+const TraceProcessor = require('../lib/tracehouse/trace-processor.js');
+
+// TraceProcessor throws generic errors, but we'd like our special localized and code-specific LHError
+// objects to be thrown instead.
+class LHTraceProcessor extends TraceProcessor {
+  /**
+   * @return {Error}
+   */
+  static createNoNavstartError() {
```

Comment:
this is a problem I've never seen before (translating error codes from one domain to another). What made you go with this method override approach vs try/catch/replacing (or .... drum roll.... catch and release)

Reply:
Catch and rethrow would be dependent on the specific string used inside `trace-processor`. I've been burned by such an approach before, so wanted to avoid it.

This method also makes it very clear which important errors can be thrown by the trace processor and gives other consumers of tracehouse the ability to override if they wish too.